### PR TITLE
fix opetee-test/dpdk/etc. for qoriq

### PR DESCRIPTION
**Pull request by ting-liu**

### CODE REVIEW

File: recipes-extended/dpdk/dpdk.inc [hunk 1]
```diff
@@ -16,6 +20,7 @@ COMPATIBLE_HOST_armv6 = 'null'
 
 COMPATIBLE_MACHINE = "(imx|qoriq)"
 
+export INSTALL_PATH = "${prefix}/share/dpdk"
```
Comment (otavio) [inline recipes-extended/dpdk/dpdk.inc:23]: This is `${localstatedir}` so please use this accordingly. No hardcoded values please.

File: recipes-extended/dpdk/dpdk.inc [hunk 2]
```diff
@@ -82,59 +87,74 @@ do_install () {
     oe_runmake O=${RTE_OUTPUT} T= install-sdk DESTDIR=${D}
 
     # install examples without openssl enabled
-    install -m 0755 -d ${D}/${datadir}/dpdk/examples
+    install -m 0755 -d ${D}${INSTALL_PATH}/examples
```
Comment (otavio) [inline recipes-extended/dpdk/dpdk.inc:90]: Use `${localstatedir}` as well as for the rest of the file.

File: recipes-extended/dpdk/dpdk.inc [hunk 3]
```diff
@@ -113,6 +113,10 @@ do_install () {
 
     # copy nxp specific cfg and scripts
     cp -rf ${S}/nxp/* ${D}${INSTALL_PATH}/
+
+    for ss in $(find ${D} -type f -name "*.py"); do
+        sed -i -e "1s,#!.*python.*,#!${USRBINPATH}/env python3," ${ss}
+    done
```
Comment (otavio) [inline recipes-extended/dpdk/dpdk.inc:119]: Use a patch.

**Comment by otavio** [comment]
@rehsack @thochstein please review.

File: recipes-extended/dpdk/dpdk.inc (hunk 1 above)
Comment (ting-liu) [reply recipes-extended/dpdk/dpdk.inc:23]: ${localstatedir} is /var, dpdk installs tools to /usr/share/dpdk. May I know why to use ${localstatedir}?

**Comment by ting-liu** [comment]
v2:
1. dpdk: override datadir instead of adding new var
2. dpdk: add a patch to use python3
3. dpdk: merge inc and bb files
4. u-boot-qoriq: merge inc and bb files

File: recipes-extended/dpdk/dpdk.inc [hunk 4]
```diff
@@ -82,59 +86,74 @@ do_install () {
     oe_runmake O=${RTE_OUTPUT} T= install-sdk DESTDIR=${D}
 
     # install examples without openssl enabled
-    install -m 0755 -d ${D}/${datadir}/dpdk/examples
+    install -m 0755 -d ${D}${datadir}/examples
```
Comment (rehsack) [inline recipes-extended/dpdk/dpdk.inc]: Generally, the structure is general -> specific, read:
${datadir}/examples/dpdk
${datadir}/doc/dpdk
...

not
${datadir}/dpdk/$(own structure)

The latter one is discouraged.

File: recipes-extended/pktgen-dpdk/pktgen-dpdk_19.12.0.bb [hunk 5]
```diff
@@ -12,9 +12,9 @@ S = "${WORKDIR}/git"
 
 DPAA_VER ?= "dpaa"
 export RTE_TARGET = "arm64-${DPAA_VER}-linuxapp-gcc"
-export RTE_SDK = "${RECIPE_SYSROOT}/usr/share/"
+export RTE_SDK = "${RECIPE_SYSROOT}${datadir}/dpdk"
```
Comment (rehsack) [inline recipes-extended/pktgen-dpdk/pktgen-dpdk_19.12.0.bb:15]: seems to interfere with another commit. In such a case, better re-order or squash together.

File: recipes-extended/dpdk/dpdk_19.11-20.04.bb [hunk 6]
```diff
@@ -12,10 +14,162 @@ SRC_URI = "git://source.codeaurora.org/external/qoriq/qoriq-components/dpdk;nobr
...
+DPDK_RTE_TARGET_x86 = "i686-native-linuxapp-gcc"
+DPDK_RTE_TARGET_armv7a = "${ARCH}-armv7a-linuxapp-gcc"
+DPDK_RTE_TARGET_armv7ve = "${ARCH}-armv7a-linuxapp-gcc"
+DPDK_RTE_TARGET ?= "${ARCH}-dpaa-linuxapp-gcc"
+export RTE_TARGET = "${DPDK_RTE_TARGET}"
+
+export RTE_OUTPUT = "${S}/${RTE_TARGET}"
+export MODULE_DIR = "/lib/modules/${KERNEL_VERSION}/kernel/drivers/net"
+
+S = "${WORKDIR}/git"
```
Comment (rehsack) [inline recipes-extended/dpdk/dpdk_19.11-20.04.bb]: dpdk.inc is a great idea. dpdk-$(VERSION).inc would be bad.
I disagree to this change

File: recipes-bsp/u-boot/u-boot-qoriq_2019.10.bb [hunk 7]
```diff
@@ -1,7 +1,25 @@
...
+"
+
+SRC_URI = "git://source.codeaurora.org/external/qoriq/qoriq-components/u-boot;protocol=https;nobranch=1"
+SRCREV= "3cd9bc39934825200822855574fc91d8276e6584"
+
+S = "${WORKDIR}/git"
+B = "${WORKDIR}/build"
+PV_append = "+fslgit"
+LOCALVERSION = "+fsl"
+
```
Comment (rehsack) [inline recipes-bsp/u-boot/u-boot-qoriq_2019.10.bb:22]: This is sane - but there should be a u-boot-qoriq-common.inc which allowes like linux-qoriq to derive a machine BSP based setup with local git forks and BSP additions.

File: recipes-extended/dpdk/dpdk.inc (hunk 1 above)
Comment (rehsack) [reply recipes-extended/dpdk/dpdk.inc]: Yeah, probably ${datadir}

Comment (otavio) [reply recipes-extended/dpdk/dpdk.inc]: `${libexecdir}` might work better.

File: recipes-extended/dpdk/dpdk.inc [hunk 8]
```diff
@@ -27,92 +27,106 @@ export  MODULE_DIR = "/lib/modules/${KERNEL_VERSION}/kernel/drivers/net"
...
+            EXTRA_LDFLAGS="-L${STAGING_LIBDIR} --hash-style=gnu -fuse-ld=bfd" \
+            EXTRA_CFLAGS="${HOST_CC_ARCH} ${TOOLCHAIN_OPTIONS} -I${STAGING_INCDIR}" \
+            CROSS="${TARGET_PREFIX}" -C ${APP} CONFIG_RTE_LIBRTE_PMD_OPENSSL=y O="${S}/examples/${temp}"
+        else
+            oe_runmake \
+            EXTRA_LDFLAGS="-L${STAGING_LIBDIR} --hash-style=gnu -fuse-ld=bfd" \
+            EXTRA_CFLAGS="${HOST_CC_ARCH} ${TOOLCHAIN_OPTIONS} -I${STAGING_INCDIR}" \
+            CROSS="${TARGET_PREFIX}" -C ${APP} CONFIG_RTE_LIBRTE_PMD_OPENSSL=y O="${S}/examples/${temp}/"
+        fi
+    done
```
[the remark this answers is not in the document]
Comment (ting-liu) [reply recipes-extended/dpdk/dpdk.inc]: I re-read the style guide. It also say "Use spaces for indentation as developers tends to use different amount of spaces per one tab." I am OK for both:)

File: recipes-extended/dpdk/dpdk.inc (hunk 4 above)
Comment (ting-liu) [reply recipes-extended/dpdk/dpdk.inc]: Year, agree with you. dpdk default to install to ${datadir}/dpdk/, just follow it.

File: recipes-extended/dpdk/dpdk_19.11-20.04.bb (hunk 6 above)
Comment (ting-liu) [reply recipes-extended/dpdk/dpdk_19.11-20.04.bb]: just merge dpdk.inc to dpdk_${VERSION}.bb, not dpdk-$(VERSION).inc. I get your point. will revert this change.

File: recipes-extended/dpdk/dpdk.inc (hunk 8 above)
[the remark this answers is not in the document]
Comment (rehsack) [reply recipes-extended/dpdk/dpdk.inc]: I got commits rejected by OE-core guys using spaces for shell code. I can only encourage to avoid style changes to spaces for shell-code.
Using one style for all layers makes live easier ;)

File: recipes-extended/dpdk/dpdk_19.11-20.04.bb (hunk 6 above)
Comment (rehsack) [reply recipes-extended/dpdk/dpdk_19.11-20.04.bb]: It not merged yet - just take it out (git rebase -i)

File: recipes-extended/dpdk/dpdk.inc [hunk 9]
```diff
@@ -61,19 +63,16 @@ do_compile () {
...
-            CROSS="${TARGET_PREFIX}" -C ${APP} CONFIG_RTE_LIBRTE_PMD_OPENSSL=y O="${S}/examples/${temp}"
-        else
-            oe_runmake \
-            EXTRA_LDFLAGS="-L${STAGING_LIBDIR} --hash-style=gnu -fuse-ld=bfd" \
-            EXTRA_CFLAGS="${HOST_CC_ARCH} ${TOOLCHAIN_OPTIONS} -I${STAGING_INCDIR}" \
-            CROSS="${TARGET_PREFIX}" -C ${APP} CONFIG_RTE_LIBRTE_PMD_OPENSSL=y O="${S}/examples/${temp}/"
-        fi
+    oe_runmake \
+    EXTRA_LDFLAGS="-L${STAGING_LIBDIR} --hash-style=gnu -fuse-ld=bfd" \
+    EXTRA_CFLAGS="${HOST_CC_ARCH} ${TOOLCHAIN_OPTIONS} -O3 -I${STAGING_INCDIR}" \
```
Comment (thochstein) [inline recipes-extended/dpdk/dpdk.inc]: You've added a hard-coded optimization here. Seems there should be a separate commit with a reason why.

File: recipes-extended/dpdk/dpdk.inc (hunk 4 above)
Comment (thochstein) [reply recipes-extended/dpdk/dpdk.inc]: Is /usr/share just wrong to begin with? /usr/share is supposed to be for architecture-independent data. Wouldn't /opt be more appropriate?

https://refspecs.linuxfoundation.org/FHS_3.0/fhs-3.0.pdf

Comment (rehsack) [reply recipes-extended/dpdk/dpdk.inc]: Then you would have to install everything qoriq related to /opt. I had a longer discussion with @rpurdie and @rossburton wrt. /opt and Yocto etc. - and it will need some fixes before it's sanely usable.

Also - dpdk isn't architecture dependent - it's possible to have different architectures using dpaa* and dpdk.

**Comment by ting-liu** [comment]
will split it to 3 pull-request for different updates.
